source/agora/consensus/state/ValidatorSet.d

```diff
@@ -805,14 +759,35 @@ public class ValidatorSet
 
         return utxos;
     }
+
+    /***************************************************************************
+
+        Calculate the lowest height for an enrollemnt for the given height
+
+        Params:
+            height = block height
+
+        Returns:
+            min height for active enrollment at height given
+
+    ***************************************************************************/
+
+    public Height minEnrollmentHeight (in Height height) const scope @safe
+        pure nothrow @nogc
+    {
+        return Height(height <= this.params.ValidatorCycle ? 0
+            : height - this.params.ValidatorCycle);
+    }
```

Comment:
Could probably be its own commit / PR too, and nice.

Reply:
Yes will extract into commit.